tests/rule_based_profiler/config/test_base.py

```diff
@@ -0,0 +1,497 @@
+    config = schema.load(data)
+    assert isinstance(config, ExpectationConfigurationBuilderConfig)
+    assert all(getattr(config, k) == v for k, v in data.items())
+
+
+def tests_expectation_configuration_builder_config_successfully_loads_with_kwargs(
+    caplog,
+):
+    data = {
+        "expectation_type": "expect_column_pair_values_A_to_be_greater_than_B",
+        "class_name": "ExpectationConfigurationBuilder",
+        "created_on": "2022-01-12",
+        "author": "Charles Dickens",
+    }
+    schema = ExpectationConfigurationBuilderConfigSchema()
+
+    with caplog.at_level(logging.INFO):
+        config = schema.load(data)
+
+    assert isinstance(config, ExpectationConfigurationBuilderConfig)
+    assert all(getattr(config, k) == v for k, v in data.items())
+    assert len(caplog.messages) == 2  # created_on & author kwargs
+
+
+def test_expectation_configuration_builder_config_unsuccessfully_loads_with_missing_required_fields():
+    data = {}
+    schema = ExpectationConfigurationBuilderConfigSchema()
+
+    with pytest.raises(ValidationError) as e:
+        schema.load(data)
+
+    assert all(
+        f"'{attr}': ['Missing data for required field.']" in str(e.value)
+        for attr in ("class_name", "expectation_type")
+    )
+
+
+def test_expectation_configuration_builder_config_successfully_dumps_to_dictionary(
+    caplog,
+):
+    schema = ExpectationConfigurationBuilderConfigSchema()
+    config = ExpectationConfigurationBuilderConfig(
+        class_name="DomainBuilder",
+        expectation_type="expect_column_values_to_be_in_set",
+        timestamp="2017-02-15|20:26:08.937881",
+    )
+
+    with caplog.at_level(logging.INFO):
+        data = schema.dump(config)
+
+    assert isinstance(data, dict)
+    assert all(getattr(config, k) == v for k, v in data.items())
+    assert "Removed 'module_name' due to null value" in caplog.messages
+
+
+def test_rule_config_successfully_loads_with_required_args():
+    data = {
+        "name": "rule_1",
+        "domain_builder": {"class_name": "DomainBuilder"},
+        "parameter_builders": [
+            {"class_name": "ParameterBuilder", "parameter_name": "my_parameter"}
+        ],
+        "expectation_configuration_builders": [
+            {
+                "class_name": "ExpectationConfigurationBuilder",
+                "expectation_type": "expect_column_pair_values_A_to_be_greater_than_B",
+            }
+        ],
+    }
+    schema = RuleConfigSchema()
+    config = schema.load(data)
+
+    assert isinstance(config, RuleConfig)
+    assert isinstance(config.domain_builder, DomainBuilderConfig)
+    assert len(config.parameter_builders) == 1 and isinstance(
+        config.parameter_builders[0], ParameterBuilderConfig
+    )
+    assert len(config.expectation_configuration_builders) == 1 and isinstance(
+        config.expectation_configuration_builders[0],
+        ExpectationConfigurationBuilderConfig,
+    )
+
+
```

Comment:
@cdkini Just making sure that we have a test or two showing that the `module_name` is optional.  Thanks!

Reply:
Each of the `dumps` test show that even though it wasn't supplied, it is in the output. See L250 (but also let me know if we should be a bit more explicit about this).